plugins/common/oauth/config.go

```diff
@@ -28,5 +34,40 @@ func (o *OAuth2Config) CreateOauth2Client(ctx context.Context, client *http.Clie
 		client = oauthConfig.Client(ctx)
 	}
 
-	return client
+	if o.CredentialsFile != "" {
+		o.TokenURL = ctx.Value("url").(string)
```

Comment:
So any set `token_url` will always be overridden with the value from the file? So why does the sample config mention to set this parameter?

Reply:
No longer overriding `token_url`

Reply:
Am I correct that it isn't used at all anymore for the case when the credentials file is set?

Reply:
We're setting token_url in our telegraf.conf. 

[TokenURL gets passed in as the audience when requesting the access token.](https://github.com/homedepot/telegraf/blob/output-cloudrun-addition/plugins/common/oauth/config.go#L37)